internal/cloudinfo/machine_types.go

```diff
@@ -54,14 +56,38 @@ type VMKey struct {
 	cloud        string
 	service      string
 	region       string
-	location     string
 	instanceType string
 }
 
+type InstanceTypeMap struct {
+	sync.RWMutex
+	internal map[VMKey]MachineDetails
+}
+
+func NewInstanceTypeMap() *InstanceTypeMap {
```

Comment:
Inject logger and error handler as input parameters to this factory method similar how it's done in `cluster.NewManager`

Reply:
logger is created from context